src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -229,9 +256,12 @@ public StrimziKafkaContainer waitForRunning() {
     protected void containerIsStarting(final InspectContainerResponse containerInfo, final boolean reused) {
         super.containerIsStarting(containerInfo, reused);
 
-        this.kafkaExposedPort = getMappedPort(KAFKA_PORT);
-
-        LOGGER.info("Mapped port: {}", kafkaExposedPort);
+        if (this.nodeRole.isBroker()) {
+            this.kafkaExposedPort = getMappedPort(KAFKA_PORT);
+            LOGGER.info("Mapped port: {}", kafkaExposedPort);
+        } else {
+            LOGGER.info("Controller-only node - no client port exposed");
```

Comment:
We don't want to expose `CONTROLLER_PORT`? How would clients interact with the controllers?

Reply:
Ah, we need to allow users to connect outside of the Docker network (I was only thinking clients within the same Docker network). I will update it the same way we do for brokers; you can check the changes. 
---
Also, I added IT, but maybe you could think ofa  better one?